app/api/Datasets.scala

```diff
@@ -2541,6 +2535,105 @@ class  Datasets @Inject()(
     })(pec)
   }
 
+  /**
+   * Enumerator to loop over all files in a dataset and return chunks for the result zip file that will be
+   * streamed to the client. The zip files are streamed and not stored on disk.
+   *
+   * @param dataset dataset from which to get teh files
+   * @param chunkSize chunk size in memory in which to buffer the stream
+   * @param compression java built in compression value. Use 0 for no compression.
+   * @param baseURL the root Clowder URL for metadata files, from original request
+   * @param user an optional user to include in metadata
+   * @param fileIDs a list of UUIDs of files in the dataset to include (i.e. marked file downloads)
+   * @param folderId a folder UUID in the dataset to include (i.e. folder download)
+   * @return Enumerator to produce array of bytes from a zipped stream containing the bytes of each file
+   *         in the dataset
+   */
+  def enumeratorFromDatasetFiles(dataset: Dataset, chunkSize: Int = 1024 * 8,
+                                 compression: Int = Deflater.DEFAULT_COMPRESSION, baseURL: String,
+                                 user : Option[User], fileIDs: Option[List[UUID]], folderId: Option[UUID])
+                                (implicit ec: ExecutionContext): Enumerator[Array[Byte]] = {
+    implicit val pec = ec.prepare()
+    val dataFolder = ""
+    val filenameMap = scala.collection.mutable.Map.empty[UUID, String]
+    val inputFiles = scala.collection.mutable.ListBuffer.empty[models.File]
+
+    // Get list of all files and folder in dataset and enforce unique names
+    fileIDs match {
+      case Some(fids) => {
+        Logger.info("Downloading only some files")
+        Logger.info(fids.toString)
+        listFilesInFolder(fids, List.empty, dataFolder, filenameMap, inputFiles)
+      }
+      case None => {
+        folderId match {
+          case Some(fid) => listFilesInFolder(List.empty, List(fid), dataFolder, filenameMap, inputFiles)
+          case None => listFilesInFolder(dataset.files, dataset.folders, dataFolder, filenameMap, inputFiles)
+        }
+      }
+    }
+
+    // create the zipfile
+    val byteArrayOutputStream = new ByteArrayOutputStream(chunkSize)
+    val zip = new ZipOutputStream(byteArrayOutputStream)
+    zip.setLevel(compression)
+
+    var file_index = 0
+    val buffer = new Array[Byte](chunkSize)
+    var is: Option[InputStream] = None
+
+    Enumerator.generateM({
+      println("getting chunk")
```

Comment:
Should we replace all the `println` with `Logger.debug` or `Logger.trace`?

Reply:
this was supposed to be deleted, guess I missed a few.